src/battle_ai_util.c

```diff
@@ -2752,8 +2752,7 @@ bool32 AI_CanPutToSleep(u32 battlerAtk, u32 battlerDef, u32 defAbility, u32 move
 
 static bool32 AI_CanPoisonType(u32 battlerAttacker, u32 battlerTarget, u32 move)
 {
-    return ((AI_DATA->abilities[battlerAttacker] == ABILITY_CORROSION && gBattleMoves[move].split == SPLIT_STATUS)
-            || !(IS_BATTLER_OF_TYPE(battlerTarget, TYPE_POISON) || IS_BATTLER_OF_TYPE(battlerTarget, TYPE_STEEL)));
+    return CanPoisonType(battlerAttacker, battlerTarget);
```

Comment:
If we're outright ignoring the move argument, shouldn't we be removing it then?

Reply:
Maybe I just remove the func then.

Reply:
removed